utils/doubly_linked_list.mli

```diff
@@ -78,6 +78,8 @@ val exists : 'a t -> f:('a -> bool) -> bool
 
 val for_all : 'a t -> f:('a -> bool) -> bool
 
+val for_all_i : 'a t -> f:(int -> 'a -> bool) -> bool
```

Comment:
For consistency with _e.g._ `mapi` [1],
the name should maybe be `for_alli`?

[1] https://github.com/ocaml-flambda/flambda-backend/blob/main/stdlib/list.mli#L186

Reply:
Renamed.